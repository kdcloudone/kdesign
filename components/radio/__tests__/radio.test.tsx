import React from 'react'
import { mount, render } from 'enzyme'
import Radio, { Group, Button } from '..'
import focusTest from '../../../tests/shared/focusTest'
import mountTest from '../../../tests/shared/mountTest'

describe('Radio', () => {
  focusTest(Radio, { refFocus: true })
  mountTest(Radio)
  mountTest(Group)
  mountTest(Button)

  it('should render correctly', () => {
    const wrapper = render(
      <Radio className="customized" style={{ color: 'red' }} radioType="square" prefixCls="myPrefix">
        Test
      </Radio>,
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('responses change events', () => {
    const onChange = jest.fn()

    const wrapper = mount(<Radio onChange={onChange}>Test</Radio>)

    wrapper.find('.kd-radio-input').simulate('change')
    expect(onChange).toHaveBeenCalled()
  })

  it('should have displayName static property', () => {
    const wrapper = mount(<Radio />)
    expect((wrapper.type() as any).displayName).toBe('Radio')
  })
})
