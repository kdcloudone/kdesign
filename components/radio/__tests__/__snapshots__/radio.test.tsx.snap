// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio should render correctly 1`] = `
<label
  class="myPrefix customized"
  style="color:red"
>
  <input
    class="myPrefix-input"
    type="radio"
  />
  <span
    class="myPrefix-text"
  >
    Test
  </span>
</label>
`;
